js/store.js

```diff
@@ -308,9 +312,10 @@ const store = {
           // show third party; it either became visible or is brand new
           thirdParty['isVisible'] = true;
           isNewThirdParty = true;
-          thirdParty['firstPartyHostnames'].forEach((firstPartyHostname) => {
-            this.addFirstPartyLink(firstPartyHostname, target);
-          });
+          thirdParty['firstPartyHostnames']
+            .forEach((firstPartyHostname) => {
+              this.addFirstPartyLink(firstPartyHostname, target);
```

Comment:
You mean make the arrow function inside `forEach` async to add an await for `addFirstPartyLink`? This would prevent adding the next link until the previous one had completed and done a write?